tools/benchmark/src/MochaCustomOutputReporter.ts

```diff
@@ -0,0 +1,193 @@
+					console.info(
+						chalk.red(
+							`Test ${test.title} in ${suite} completed with status '${test.state}' after reporting data.`,
+						),
+					);
+					benchmarkResults.aborted = true;
+				}
+
+				if (isChildProcess) {
+					// Write the data to stdout so the parent process can collect it.
+					console.info(JSON.stringify(benchmarkResults));
+				} else {
+					let suiteData = this.inProgressSuites.get(suite);
+					if (suiteData === undefined) {
+						suiteData = [];
+						this.inProgressSuites.set(suite, suiteData);
+					}
+					suiteData.push([getName(test.title), benchmarkResults]);
+				}
+			})
+			.on(Runner.constants.EVENT_SUITE_END, (suite: Suite) => {
+				if (!isChildProcess) {
+					const suiteName = getSuiteName(suite);
+					const suiteData = this.inProgressSuites.get(suiteName);
+					if (suiteData === undefined) {
+						return;
+					}
+					console.log(`\n${chalk.bold(suiteName)}`);
+
+					const table = new Table();
+					const failedTests = new Array<[string, CustomBenchmarkResults]>();
+
+					for (const [testName, testData] of suiteData) {
+						if (testData.aborted) {
+							table.cell("status", `${pad(4)}${chalk.red("×")}`);
+							failedTests.push([testName, testData]);
+						} else {
+							table.cell("status", `${pad(4)}${chalk.green("✔")}`);
+						}
+						table.cell("name", chalk.italic(testName));
+						if (!testData.aborted) {
+							for (const [key, value] of Object.entries(
+								testData.customMeasurements,
+							)) {
+								table.cell(key, value.toString(), Table.padLeft);
+							}
+						}
+						table.newRow();
+					}
+
+					console.log(`${table.toString()}`);
+					if (failedTests.length > 0) {
+						console.log(
+							"------------------------------------------------------",
+							`\n${chalk.red("ERRORS:")}`,
+						);
+						for (const [testName, testData] of failedTests) {
+							console.log(`\n${chalk.red(testName)}`, "\n", testData.error);
+						}
+					}
+
+					this.writeCompletedBenchmarks(suiteName);
+					this.inProgressSuites.delete(suiteName);
+				}
+			})
+			.once(Runner.constants.EVENT_RUN_END, () => {});
+	}
+
+	private writeCompletedBenchmarks(suiteName: string): string {
+		const outputFriendlyBenchmarks: unknown[] = [];
+		const suiteData = this.inProgressSuites.get(suiteName);
+		if (suiteData !== undefined) {
+			for (const [testName, testData] of suiteData) {
+				if (testData.aborted) {
+					break;
+				}
+				outputFriendlyBenchmarks.push({
+					testName,
+					testData,
+				});
+			}
+		}
+
+		// Use the suite name as a filename, but first replace non-alphanumerics with underscores
+		const suiteNameEscaped: string = suiteName.replace(/[^\da-z]/gi, "_");
+		const outputContentString: string = JSON.stringify(
+			{ suiteName, tests: outputFriendlyBenchmarks },
+			undefined,
+			4,
+		);
+
+		// If changing this or the result file logic in general,
```

Comment:
I saw this comment above as well, should it be abstracted to a NOTE in the function?

Reply:
This file is no longer needed. Remove it now.